torch/nn/functional.py

```diff
@@ -1777,8 +1777,24 @@ def multilabel_soft_margin_loss(input, target, weight=None, size_average=None,
     """
     if size_average is not None or reduce is not None:
         reduction = _Reduction.legacy_get_string(size_average, reduce)
-    input = torch.sigmoid(input)
-    return binary_cross_entropy(input, target, weight, None, None, reduction)
+
+    loss = - (target * logsigmoid(input) + (1 - target) * logsigmoid(-input))
+
+    if weight is not None:
+        loss = loss * weight
+
+    if reduction != 'none':
+        if reduction == 'sum':
+            return loss.sum()
+        else:  # global mean
+            return loss.mean()
```

Comment:
Let's resolve #9301 and #10034 (both are the same issue) by changing the behavior to match the docs (divide by `size(0)`).